source/guide_keycloak.rst

```diff
@@ -0,0 +1,274 @@
+.. highlight:: console
+
+.. author:: Julius Künzel
+
+.. tag:: web
+.. tag:: sso
+
+.. spelling::
+    Keycloak
+
+.. sidebar:: Logo
+
+  .. image:: _static/images/keycloak.svg
+      :align: center
+
+########
+Keycloak
+########
+
+.. tag_list::
+
+Keycloak is an open source identity and access management system which allows single sign-on for web applications and services.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * :manual:`supervisord <daemons-supervisord>`
+  * :manual:`Backends <web-backends>`
+
+License
+=======
+
+Keycloak is released under the `Apache 2.0 license`_.
+
+Prerequisites
+=============
+
+Your URL needs to be setup:
+
+.. include:: includes/web-domain-list.rst
+
+Installation
+============
+
+Download Keycloak
+-----------------
+
+Check the Keycloak_ website or `Github Repository`_ for the latest release and copy the download link to the .tar.gz file. Then use ``wget`` to download it. Replace the URL with the one you just got from GitHub.
+
+::
+
+ [isabell@stardust ~]$ wget https://github.com/keycloak/keycloak/releases/download/15.0.2/keycloak-15.0.2.tar.gz
+ […]
+ Saving to: ‘keycloak-15.0.2.tar.gz’
+
+ 100%[======================================>] 253,994,058 51,5MB/s   in 4,8s
+
+ 2021-11-19 22:36:18 (50,7 MB/s) - ‘keycloak-15.0.2.tar.gz’ saved [253994058/253994058]
+ [isabell@stardust ~]$
+
+Extract the archive
+-------------------
+
+Use ``tar`` to extract the archive and delete the archive. Replace the version in the archive file name with the one you just downloaded.
+
+::
+
+ [isabell@stardust ~]$ tar -xvzf keycloak-15.0.2.tar.gz
+ keycloak-15.0.2
+ [isabell@stardust ~]$
+
+You can now delete the archive:
+
+::
+
+ [isabell@stardust ~]$ rm keycloak-15.0.2.tar.gz
+ [isabell@stardust ~]$
+
+Configuration
+=============
+
+Change the configuration
+------------------------
+
+Find the following block in file :file:`standalone/configuration/standalone.xml` and add the ``proxy-address-forwarding="true"`` attribute to ``<http-listener>`` element under ``<server>``.
+
+.. code-block:: xml
+ :emphasize-lines: 4
+
+ <subsystem xmlns="urn:jboss:domain:undertow:10.0" default-server="default-server" default-virtual-host="default-host" default-servlet-container="default" default-security-domain="other" statistics-enabled="${wildfly.undertow.statistics-enabled:${wildfly.statistics-enabled:false}}">
+     <buffer-cache name="default"/>
+     <server name="default-server">
+         <http-listener name="default" socket-binding="http" redirect-socket="https" enable-http2="true" read-timeout="30000" proxy-address-forwarding="true"/>
```

Comment:
This snippet looks different than the one provided in the tarball (I think it's mostly the added `read-timeout` attribute):
```xml
<http-listener name="default" socket-binding="http" redirect-socket="https" enable-http2="true"/>
<https-listener name="https" socket-binding="https" security-realm="ApplicationRealm" enable-http2="true"/>
```
Can you take another look and align them, please?